File: backend/internal/constants/billing.go
package constants

// BillingType 费用类型枚举。
const (
	BillingTypeAttorneyFee = "attorney_fee"
	BillingTypeCourtFee    = "court_fee"
	BillingTypeTravelFee   = "travel_fee"
	BillingTypeOther       = "other"
)

// BillingTypeValues 全部费用类型值。
var BillingTypeValues = []string{BillingTypeAttorneyFee, BillingTypeCourtFee, BillingTypeTravelFee, BillingTypeOther}

// BillingStatus 账单状态枚举。
const (
	BillingStatusPending  = "pending"
	BillingStatusPaid     = "paid"
	BillingStatusInvoiced = "invoiced"
	BillingStatusVoid     = "void"
)

// BillingStatusValues 全部账单状态值。
var BillingStatusValues = []string{BillingStatusPending, BillingStatusPaid, BillingStatusInvoiced, BillingStatusVoid}

// IsValidBillingType 校验费用类型。
func IsValidBillingType(s string) bool {
	for _, v := range BillingTypeValues {
		if v == s {
			return true
		}
	}
	return false
}

// IsValidBillingStatus 校验账单状态。
func IsValidBillingStatus(s string) bool {
	for _, v := range BillingStatusValues {
		if v == s {
			return true
		}
	}
	return false
}

// DocumentFileType 文档类型枚举。
const (
	DocTypeComplaint = "complaint"
	DocTypeDefense   = "defense"
	DocTypeEvidence  = "evidence"
	DocTypeJudgment  = "judgment"
	DocTypeContract  = "contract"
	DocTypeOther     = "other"
)

// DocumentFileTypeValues 全部文档类型值。
var DocumentFileTypeValues = []string{DocTypeComplaint, DocTypeDefense, DocTypeEvidence, DocTypeJudgment, DocTypeContract, DocTypeOther}

File: frontend/src/constants/billing.ts
// 费用类型/账单状态枚举（与后端 backend/internal/constants/billing.go 保持一致）
export const BillingType = {
  ATTORNEY_FEE: 'attorney_fee',
  COURT_FEE: 'court_fee',
  TRAVEL_FEE: 'travel_fee',
  OTHER: 'other',
} as const

export const BillingTypeText: Record<string, string> = {
  [BillingType.ATTORNEY_FEE]: '律师费',
  [BillingType.COURT_FEE]: '诉讼费',
  [BillingType.TRAVEL_FEE]: '差旅费',
  [BillingType.OTHER]: '其他',
}

export const BillingStatus = {
  PENDING: 'pending',
  PAID: 'paid',
  INVOICED: 'invoiced',
  VOID: 'void',
} as const

export const BillingStatusText: Record<string, string> = {
  [BillingStatus.PENDING]: '待支付',
  [BillingStatus.PAID]: '已支付',
  [BillingStatus.INVOICED]: '已开票',
  [BillingStatus.VOID]: '已作废',
}

export const BillingTypeOptions = Object.entries(BillingTypeText).map(([value, label]) => ({ label, value }))
export const BillingStatusOptions = Object.entries(BillingStatusText).map(([value, label]) => ({ label, value }))
